### PR TITLE
Import all the authentication plugins from client-go

### PR DESCRIPTION
**Pull request by jdartigalongue**
/assign @rrati 
/assign @mphammer

### CODE REVIEW

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/331#" title="Author self-approved">jdartigalongue</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by operator-r2d2** [comment]
@rrati: once the present PR merges, I will cherry-pick it on top of release-2019.4.x in a new PR and assign it to you.

<details>

In response to [this](https://github.com/blackducksoftware/synopsys-operator/pull/331#issuecomment-491849971):

>/cherrypick release-2019.4.x


Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by operator-r2d2** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/331#" title="Author self-approved">jdartigalongue</a>*, *<a href="https://github.com/blackducksoftware/synopsys-operator/pull/331#pullrequestreview-236719032" title="LGTM">rrati</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=blackducksoftware%2Fsynopsys-operator).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/blackducksoftware/synopsys-operator/blob/master/OWNERS)~~ [jdartigalongue,rrati]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/synopsysctl/cmd_root.go [hunk 1]
```diff
@@ -53,7 +53,9 @@ var rootCmd = &cobra.Command{
 		}
 		log.SetLevel(lvl)
 		// Sets kubeconfig and initializes resource client libraries
-		setResourceClients()
+		if err := setResourceClients(); err != nil {
+			return err
```
Comment (mphammer) [inline pkg/synopsysctl/cmd_root.go]: Returning an error here will print the `synopsysctl --help` message. Consider logging an error and returning `nil`.   Also, returning an error will prevent synopsysctl from running if you are not connected to a cluster and there is an error from getting kubeconfig/kubeclient.

**Pull request closed without merging**